src/tools/run-make-support/src/symbols.rs

```diff
@@ -42,3 +42,29 @@ pub fn any_symbol_contains(path: impl AsRef<Path>, substrings: &[&str]) -> bool
         false
     })
 }
+
+/// Check if an object file contains *all* of the given symbols.
+///
+/// The symbol names must match exactly.
+///
+/// Panics if `path` is not a valid object file readable by the current user.
+pub fn contains_exact_symbols(path: impl AsRef<Path>, symbol_names: &[&str]) -> bool {
```

Comment:
Suggestion: having this return a bool but then also have output logging is weird. We probably want to have a `read_symbols(path)` return something like `Vec<OsString>` or  `Vec<Vec<u8>>` (pretty sure symbols don't need to be UTF-8), and then have an assertion helper based on that:

```rs
fn assert_symbols_match_exactly(expected_symbols: &[&OsStr], found_symbols: &[&OsStr]) {
    assert_eq!(expected_symbols, found_symbols, "symbols do not match exactly");
}
```

or something more generic, like `assert_byte_strings_are_equal`.

(or some customized equality assertion to not show u8 slice debug repr)

Reply:
Problem: am I understanding the logic here is that for a given object file, and a given list of expected symbols, we want the object file to contain the symbol *at least once*? Or does the test actually want *exact* counts of occurrences of symbols? AFAICT, the logic here only checks that a given expected symbol occurs *at least once*, whereas the logic in the original Makefile checks that a given expected symbol occurs *exactly once*, so the logic is not equivalent from what I can tell.

Reply:
the previous makefile wasn't checking for exact symbol names though.

"accidentally generating duplicate symbols and somehow not encountering a link error" doesn't seem like a very probable failure mode, and if it is likely, then i would rather approach that with a helper that asserts a file does not contain *any* duplicate symbols, rather than checking the counts of specific symbols.

i should probably move the debug output into assertion helpers though, the current behavior is based off of the cat-and-grep script, but it's a bit confusing, and we can do better.